packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart

```diff
@@ -47,11 +41,15 @@ class FirebaseAuth extends FirebasePluginPlatform {
   }
 
   /// Returns an instance using a specified [FirebaseApp].
-  /// Note that persistence can only be used on Web and is not supported on other platforms.
-  factory FirebaseAuth.instanceFor(
-      {required FirebaseApp app, Persistence? persistence}) {
+  factory FirebaseAuth.instanceFor({
+    required FirebaseApp app,
+    @Deprecated(
+      'Will be removed in future release. Use setPersistence() instead.',
+    )
+    Persistence? persistence,
+  }) {
     return _firebaseAuthInstances.putIfAbsent(app.name, () {
-      return FirebaseAuth._(app: app, persistence: persistence);
+      return FirebaseAuth._(app: app);
```

Comment:
If we remove the behavior, it's not a deprecation but a straight removal (breaking) even if we are warning the user.
we should just put the warning now, and remove the behavior at the next breaking release